<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ./stretch_urdf/SE3/xacro/stretch_description_SE3_eoa_wrist_dw3_tool_sg3.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="stretch">
  <link name="base_link"></link>
  <link name="link_x"></link>
  <joint name="joint_base_x" type="prismatic">
    <parent link="base_link"/>
    <child link="link_x"/>
    <axis xyz="1 0 0"></axis>
    <limit lower="-1.0e16" upper="1.0e16" effort="100" velocity="1.0"/>
  </joint>
  <link name="link_y"></link>
  <joint name="joint_base_y" type="prismatic">
    <parent link="link_x"/>
    <child link="link_y"/>
    <axis xyz="0 1 0"></axis>
    <limit lower="-1.0e16" upper="1.0e16" effort="100" velocity="1.0"/>
  </joint>
  <link name="link_base">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.087526 -0.001626 0.081009"/>
      <mass value="17.384389"/>
      <inertia ixx="0.160002" ixy="0.006758" ixz="0.004621" iyy="0.138068" iyz="0.002208" izz="0.228992"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/base_link.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/base_link_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_base_z" type="continuous">
    <parent link="link_y"/>
    <child link="link_base"/>
    <axis xyz="0 0 1"></axis>
  </joint>
  <link name="link_mast">
    <inertial>
      <origin rpy="0 0 0" xyz="0.000337 0.722201 0.002652"/>
      <mass value="1.764017"/>
      <inertia ixx="0.291286" ixy="0.004149" ixz="-0.000174" iyy="0.001226" iyz="0.000354" izz="0.291429"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_mast.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_mast_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_mast" type="fixed">
    <origin rpy="1.5708 0 4.8006E-17" xyz="-0.067 0.135 0.0284"/>
    <parent link="link_base"/>
    <child link="link_mast"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="link_lift">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.020369 0.03438 0.02236"/>
      <mass value="1.987167"/>
      <inertia ixx="0.006518" ixy="-0.001539" ixz="0.000371" iyy="0.004464" iyz="0.000392" izz="0.008092"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_lift.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_lift_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_lift" type="prismatic">
    <origin rpy="-1.5708 1.5708 0" xyz="-0.037385 0.1666 0"/>
    <parent link="link_mast"/>
    <child link="link_lift"/>
    <axis xyz="0 0 1"/>
    <!-- for now: hand copied range_m: from lift: from ~/repos/stretch_fleet/stretch-re1-1001/stretch_re1_factory_params.yaml -->
    <!--<limit effort="100" lower="0.0" upper="1.095" velocity="1.0"/>-->
    <!-- copied value did not reach the top of mesh model with GUI sliders and RViz -->
    <limit effort="100" lower="0.0" upper="1.1" velocity="1.0"/>
  </joint>
  <link name="link_arm_l4">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.000168 -0.000254 -0.125235"/>
      <mass value="0.235626"/>
      <inertia ixx="0.001518" ixy="-1E-06" ixz="3E-06" iyy="0.001519" iyz="3E-06" izz="0.000176"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_arm_l4.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.79216 0.81961 0.93333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_arm_l4_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_arm_l4" type="fixed">
    <origin rpy="1.5708 2.4721E-15 -1.5708" xyz="-0.2547 0 0"/>
    <parent link="link_lift"/>
    <child link="link_arm_l4"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="link_arm_l0">
    <inertial>
      <origin rpy="0 0 0" xyz="0.033681 -0.000847 -0.031723"/>
      <mass value="0.427734"/>
      <inertia ixx="0.001689" ixy="-2.8E-05" ixz="0.000483" iyy="0.002107" iyz="-1.1E-05" izz="0.000571"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_arm_l0.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_arm_l0_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_arm" type="prismatic">
    <origin rpy="7.63746778746202E-30 -3.80121128864402E-15 2.62707547767438E-15" xyz="0 0 -0.0137499999991968"/>
    <parent link="link_arm_l4"/>
    <child link="link_arm_l0"/>
    <axis xyz="0 0 1"/>
    <!-- 0.13 = 0.52/4-->
    <limit effort="100" lower="0.0" upper="0.52" velocity="1.0"/>
  </joint>
  <link name="link_wrist_yaw">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -3.9E-05 -0.016495"/>
      <mass value="0.054422"/>
      <inertia ixx="9E-06" ixy="0" ixz="0" iyy="9E-06" iyz="0" izz="7E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_wrist_yaw.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.898039215686275 0.917647058823529 0.929411764705882 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_wrist_yaw_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_wrist_yaw" type="revolute">
    <origin rpy="-1.5707963267949 0 3.14159265358978" xyz="0.0830000000000654 -0.0307500000000129 0"/>
    <parent link="link_arm_l0"/>
    <child link="link_wrist_yaw"/>
    <axis xyz="0 0 -1"/>
    <!--
      stowed to front ~225 deg: 3.15159 x 1.25 = 3.9395
      using 4.0
      stowed to back ~100 deg:  100 / 180 x 3.14159 = 1.7453
      using -1.75
    -->
    <limit effort="100" lower="-1.75" upper="4.0" velocity="1.0"/>
  </joint>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1"/>
  </material>
  <material name="aluminum_d405">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="plastic_d405">
    <color rgba="0.1 0.1 0.1 1"/>
  </material>
  <link name="link_wrist_yaw_bottom">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0165399153795968 -0.0446328107020906 -0.0281087384763343"/>
      <mass value="0.144548635554334"/>
      <inertia ixx="4.56525957614798E-05" ixy="-1.08217417869179E-06" ixz="-1.44384040297111E-08" iyy="7.22875259104698E-05" iyz="3.40820143544021E-09" izz="8.76544308203217E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_DW3_wrist_yaw_bottom.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_DW3_wrist_yaw_bottom_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_wrist_yaw_bottom" type="fixed">
    <origin rpy="-3.14159265358979 0 -1.5707963267949" xyz="0 0 0"/>
    <parent link="link_wrist_yaw"/>
    <child link="link_wrist_yaw_bottom"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="link_wrist_pitch">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.000711479645697954 -0.0147442681465562 0.0195140764319514"/>
      <mass value="0.116590512374845"/>
      <inertia ixx="5.76204624498191E-05" ixy="-7.78647509857216E-07" ixz="-2.86646997376633E-08" iyy="3.5844353951621E-05" iyz="1.06445058968725E-06" izz="6.36355321976534E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_DW3_wrist_pitch.STL"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_DW3_wrist_pitch_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_wrist_pitch" type="revolute">
    <origin rpy="1.5708 0 3.1416" xyz="0 -0.019455 -0.0305"/>
    <parent link="link_wrist_yaw_bottom"/>
    <child link="link_wrist_pitch"/>
    <axis xyz="0 0 -1"/>
    <limit effort="100" lower="-1.57" upper="0.56" velocity="1.0"/>
  </joint>
  <link name="link_wrist_roll">
    <inertial>
      <origin rpy="0 0 0" xyz="0.000282866582415386 0.000557472767112266 0.0196635450480892"/>
      <mass value="0.0221916689638078"/>
      <inertia ixx="2.20608319376175E-06" ixy="3.01742816614342E-09" ixz="-1.83537060988685E-08" iyy="2.91436292743443E-06" iyz="-4.5509705441618E-08" izz="4.83731409646016E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_DW3_wrist_roll.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.898039215686275 0.917647058823529 0.929411764705882 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_DW3_wrist_roll_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_wrist_roll" type="revolute">
    <origin rpy="3.1416 1.5708 0" xyz="-0.018859 -0.024 0.01955"/>
    <parent link="link_wrist_pitch"/>
    <child link="link_wrist_roll"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-3.14" upper="3.14" velocity="1.0"/>
  </joint>
  <link name="link_gripper_s3_body">
    <inertial>
      <origin rpy="0 0 0" xyz="0.000296364207693828 0.00742489472460528 0.0440194148825503"/>
      <mass value="0.288995995074102"/>
      <inertia ixx="0.000482695285602815" ixy="-3.02029982967392E-06" ixz="6.73858128561175E-07" iyy="0.000213430693267808" iyz="-1.95797907844847E-05" izz="0.000377638305434195"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_SG3_gripper_body.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://stretch_description/meshes/link_SG3_gripper_body_collision.STL"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_gripper_s3_body" type="fixed">
    <origin rpy="0 0 -3.14159265358975" xyz="0 0 0.0209999999993159"/>
    <parent link="link_wrist_roll"/>
    <child link="link_gripper_s3_body"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="link_grasp_center">
  </link>
  <joint name="joint_grasp_center" type="fixed">
    <origin rpy="-1.5707963267949 -1.5707963267948 0" xyz="0 0 0.23"/>
    <parent link="link_gripper_s3_body"/>
    <child link="link_grasp_center"/>
  </joint>
</robot>
